iai-callgrind-runner/src/runner/tool/logfile_parser.rs

```diff
@@ -44,23 +46,97 @@ pub struct LogfileParser {
 #[derive(Debug, Clone)]
 pub struct LogfileSummary {
```

Comment:
A single `LogfileSummary` should only contain the parsing results of a single logfile. And instead of storing the `CostsSummary` you can store the optional `Costs` in it. The `CostsSummary` shouldn't be calculated as a parsing result. This should also make the `OldLogfileSummary` struct redundant.